python/ql/src/experimental/Security/CWE-352/DjangoCSRF.ql

```diff
@@ -0,0 +1,36 @@
+/**
+ * @name Disabled Django CSRF protection
+ * @description No csrf Middleware/ Decorator to protect Django application from CSRF
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id py/django-csrf
+ * @tags security
+ *       external/cwe/cwe-352
+ */
+
+import python
+import semmle.python.strings
+
+from GlobalVariable g
+where
+  /*
+   * Regex matching MIDDLEWARE & MIDDLEWARE_CLASSES(backwards compatibility)
+   * Checks if there is a GlobalVariable(List/Tuple) named MIDDLEWARE or MIDDLEWARE_CLASES
+   * and contains the middleware string
+   */
+
+  g.getId().regexpMatch("MIDDLEWARE(_CLASSES)?") and
+  not exists(List l |
+    l.getParentNode().getAChildNode().toString().regexpMatch(g.getId()) and
+    exists(StrConst mw | mw.getS().regexpMatch(".*csrf.*") and l.contains(mw))
```

Comment:
Can we use the whole string for the middleware here? (again, instead of regexp matching)